src/main/java/fi/hsl/transitdata/pubtransredisconnect/Main.java

```diff
@@ -104,8 +92,13 @@ private void process() {
         if (!processingActive.getAndSet(true)) {
             log.info("Fetching data");
             try (Connection connection = DriverManager.getConnection(connectionString)) {
-                queryJourneyData(connection, jedis);
-                queryStopData(connection, jedis);
+                final QueryProcessor journeyQueryProcessor = new QueryProcessor(connection, new JourneyResultSetProcessor(jedis, redisTTLInSeconds));
+                final QueryProcessor stopQueryProcessor = new QueryProcessor(connection, new StopResultSetProcessor(jedis, redisTTLInSeconds));
+                final QueryProcessor metroJourneyQueryProcessor = new QueryProcessor(connection, new MetroJourneyResultSetProcessor(jedis, redisTTLInSeconds));
+
+                journeyQueryProcessor.executeQuery(QueryUtil.getJourneyQuery(queryHistoryInDays, queryFutureInDays));
+                stopQueryProcessor.executeQuery(QueryUtil.getStopQuery());
+                metroJourneyQueryProcessor.executeQuery(QueryUtil.getMetroJourneyQuery(queryHistoryInDays, queryFutureInDays));
```

Comment:
This is probably a matter of opinion so no need to change anything, more like a comment/thought/discussion:
- the logic of processing the results is abstracted to a class but the queries are not. those could be part of that same logical encapsulation..?
- creating new processors seems funny since you could easily reuse them. Of course this works fine also, no need to change. 

Anyhow it's a good idea to separate those into classes 👍

Reply:
It sounds good so I will change this.